### ISSUE TITLE
Allow window to be transparent

### ISSUE THREAD

**Issue by louisgjohnson** [bevyengine/bevy#3032]
## What problem does this solve or what need does it fill?

For an example this feature would help users build desktop widgets or something similar to this game: https://samperson.itch.io/desktop-goose

## What solution would you like?

A way to set the transparency of the window on application init or in a system

## What alternative(s) have you considered?

I've tried accessing the `ResMut<Windows>` but no api is exposed to modify the winit window transparency 

## Additional context

It seems that winit allows the transparency to be set via the WindowBuilder but not after the creation of the window.
winit docs:
https://docs.rs/winit/0.25.0/winit/window/struct.WindowBuilder.html#method.with_transparent
https://docs.rs/winit/0.25.0/winit/window/struct.WindowAttributes.html#structfield.transparent

**Comment by rezural** [comment]
@louisgjohnson I can confirm this works on Ubuntu 20.04 with nvidia 1070

**Comment by AsessionGuy** [comment]
> I think I found a workaround on Linux, forcing `winit` to run on it's master branch fixes this, or at least just on my Manjaro Linux machine with Nvidia proprietary drivers. I'm assuming that this now works because of this PR: [rust-windowing/winit#2006](https://github.com/rust-windowing/winit/pull/2006)

Could you elaborate more on this please?

**Comment by Supermath101** [comment]
> > I think I found a workaround on Linux, forcing `winit` to run on it's master branch fixes this, or at least just on my Manjaro Linux machine with Nvidia proprietary drivers. I'm assuming that this now works because of this PR: [rust-windowing/winit#2006](https://github.com/rust-windowing/winit/pull/2006)
> 
> Could you elaborate more on this please?

TLDR: What I said is irrelevant in current Bevy versions.

This was a while ago, not sure exactly what I meant. IIRC, I might've ran some code that "implemented" PR #3033/#3105, but without directly modifying Bevy's source. Then, I ran into another issue. There was a bug in the latest (at the time) `winit` version on crates.io. This bug was a blocker in getting transparency to work. Once the `winit` version was set to the master branch of the git repo, I recall that window transparency was working finally. 

Since making that comment, the fix that previously required using the master branch of `winit` should, by now, be in a crates.io release. Also, the PRs from Bevy mentioned previously and from `winit` mentioned in the original commment were both merged ages ago, and many changes have happened since then. For all those reasons, this entire conversation is no longer relevant in current Bevy versions.

**Comment by AsessionGuy** [comment]
> TLDR: What I said is irrelevant in current Bevy versions.
> 
> This was a while ago, not sure exactly what I meant. IIRC, I might've ran some code that "implemented" PR #3033/#3105, but without directly modifying Bevy's source. Then, I ran into another issue. There was a bug in the latest (at the time) `winit` version on crates.io. This bug was a blocker in getting transparency to work. Once the `winit` version was set to the master branch of the git repo, I recall that window transparency was working finally.
> 
> Since making that comment, the fix that previously required using the master branch of `winit` should, by now, be in a crates.io release. Also, the PRs from Bevy mentioned previously and from `winit` mentioned in the original commment were both merged ages ago, and many changes have happened since then. For all those reasons, this entire conversation is no longer relevant in current Bevy versions.

Thanks for your response. After asking for more details I thought that this might be outdated so I started a discussion asking for help to get a transparent window working on linux (either x11 or wayland) using current bevy version. Honestly this issue makes me feel sad about the possible regression.
